### PR TITLE
Notice when paths from config file are overridden by command arguments

### PR DESCRIPTION
**Pull request by julienfalque**
Implements #2912.

I didn't add test as the command isn't tested at all, I plan to do it properly in a separate PR.

### CODE REVIEW

File: src/Console/ConfigurationResolver.php [hunk 1]
```diff
@@ -493,6 +498,14 @@ public function shouldStopOnViolation()
     }
 
     /**
+     * @return null|bool
+     */
+    public function configFinderIsOverridden()
+    {
+        return $this->configFinderOverridden;
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: this shall never return null. 
all properties here are nullable, but getters are not providing nulls. null means here only "lazy computed"

Comment (julienfalque) [inline src/Console/ConfigurationResolver.php]: Updated.

**Comment by keradus** [comment]
please dont amend

File: src/Console/ConfigurationResolver.php [hunk 2]
```diff
@@ -493,6 +498,14 @@ public function shouldStopOnViolation()
     }
 
     /**
+     * @return bool
+     */
+    public function configFinderIsOverridden()
+    {
+        return $this->configFinderOverridden;
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: if one wont call `getFinder` first, this will wrongly be always `false`

Comment (keradus) [inline src/Console/ConfigurationResolver.php]: hm... lack of tests for it ;P

File: src/Console/ConfigurationResolver.php [hunk 3]
```diff
@@ -99,6 +99,11 @@
     private $fixers;
 
     /**
+     * @var bool
+     */
+    private $configFinderOverridden = false;
```
[the remark this answers is not in the document]
Comment (julienfalque) [reply src/Console/ConfigurationResolver.php]: Updated.

File: src/Console/ConfigurationResolver.php (hunk 2 above)
Comment (julienfalque) [reply src/Console/ConfigurationResolver.php]: Test added.

File: src/Console/Command/FixCommand.php [hunk 4]
```diff
@@ -174,6 +174,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $progressType = $resolver->getProgress();
         $finder = $resolver->getFinder();
 
+        if (null !== $stdErr && $resolver->configFinderIsOverridden()) {
+            $stdErr->writeln(
+                sprintf($stdErr->isDecorated() ? '<bg=yellow;fg=black;>%s</>' : '%s', 'Paths from configuration file have been overridden by paths provided as command arguments.')
```
Comment (keradus) [inline src/Console/Command/FixCommand.php]: added `->isDecorated` for now for consistency.
we need to analyse do we still need to check `->isDecorated` on our side (and maybe drop it globally, not only for this one message)

**Comment by keradus** [comment]
Thank you @julienfalque.

**Comment by julienfalque** [comment]
Thanks :)

**Pull request closed without merging**